src/components/ids-treemap/ids-treemap.js

```diff
@@ -0,0 +1,428 @@
+          ${this.result.map((rect) => this.templateGroups(rect)).join('')}
+        </svg>
+      `;
+    }
+    return svg;
+  }
+
+  /**
+   * Render the group markup
+   * @param {object} rect item in the result array
+   * @returns {*} svg group
+   * @memberof IdsTreeMap
+   */
+  templateGroups(rect) {
+    const textOffset = 8;
+
+    return `
+      <g
+        fill=${rect.data.color}
+        class="treemap-rect"
+      >
+        <rect
+          x=${rect.x}
+          y=${rect.y}
+          width=${rect.width}
+          height=${rect.height}
+        >
+        </rect>
+        <text
+          fill="white"
+          x="${rect.x + textOffset * 2}"
+          y="${rect.y + textOffset * 3}"
+          stroke-width="0"
+        >
+          ${rect.data.text}
+        </text>
+        <text
+          fill="white"
+          x="${rect.x + textOffset * 2}"
+          y="${rect.y + textOffset * 6}"
+          stroke-width="0"
+        >
+          ${rect.data.label}
+        </text>
+      </g>
+    `;
+  }
+
+  /**
+   * Render the title markup
+   * @returns {*} Title banner markup
+   * @memberof IdsTreeMap
+   */
+  templateTitle() {
+    return `
+      <ids-text type="span" font-weight="bold">
+        ${this.title !== null ? this.title : 'Add Treemap Title'}
+      </ids-text>`;
+  }
+
+  /**
+   * Render the treemap by applying the template
+   * @private
+   */
+  render() {
+    super.render();
+  }
+
+  /**
+   * Get max number
+   * @param {Array} array row
+   * @returns {Array} max number
+   * @memberof IdsTreeMap
+   * @private
+   */
+  getMaximum = (array) => Math.max(...array);
+
+  /**
+   * Get min number
+   * @param {Array} array row
+   * @returns {Array} min number
+   * @memberof IdsTreeMap
+   * @private
+   */
+  getMinimum = (array) => Math.min(...array);
+
+  /**
+   * Sum Reducer
+   * @param {Array} acc row
+   * @param {Array} cur row
+   * @returns {Array} reduced array
+   * @memberof IdsTreeMap
+   * @private
+   */
+  sumReducer = (acc, cur) => acc + cur;
+
+  /**
+   * Round Value
+   * @param {Array} number row
+   * @returns {Array} round value array
+   * @memberof IdsTreeMap
+   * @private
+   */
+  roundValue = (number) => Math.max(Math.round(number * 100) / 100, 0);
+
+  /**
+   * Validate the treemap object.
+   * @param {object} obj { data, height }
+   * @param {Array} obj.data array that contains the treemap block definitions
+   * @param {number} obj.height total hieght of the treemap
+   * @memberof IdsTreeMap
+   * @private
+   */
+  validateArguments = ({ data, height }) => {
+    if (!height || typeof height !== 'number' || height < 0) {
+      throw new Error('You need to specify the height of your treemap');
+    }
+    if (!data || !Array.isArray(data) || data.length === 0 || !data.every((dataPoint) => Object.prototype.hasOwnProperty.call(dataPoint, 'value') && typeof dataPoint.value === 'number' && dataPoint.value >= 0 && !Number.isNaN(dataPoint.value))) {
```

Comment:
What exactly does this do?

What I got from it was 
- it checks if data is an array
- checks to make sure data length > 0
- checks that each item in data has a key named `value`
- each key is a number > 0

Reply:
```suggestion
    if (!Array.isArray(data) || data.length === 0 || !data.every((dataPoint) => Object.prototype.hasOwnProperty.call(dataPoint, 'value') && typeof dataPoint.value === 'number' && dataPoint.value >= 0)) {
```

If so, can get rid of redundant statements